controller/health.go

```diff
@@ -24,6 +25,9 @@ func setApplicationHealth(resources []managedResource, statuses []appv1.Resource
 		if res.Target != nil && hookutil.Skip(res.Target) {
 			continue
 		}
+		if res.Target != nil && res.Target.GetAnnotations() != nil && res.Target.GetAnnotations()[common.AnnotationIgnoreHealthCheck] == "true" {
```

Comment:
Are the resources here all app resources or only immediate children?

Reply:
Only immediate childre. As per [documentation](https://argo-cd.readthedocs.io/en/stable/operator-manual/health/#health-checks): `An Argo CD App's health is inferred from the health of its immediate child resources`